frontend/app/controllers/Healthcheck.scala

```diff
@@ -1,34 +1,70 @@
 package controllers
 
-import com.github.nscala_time.time.Imports._
+import akka.agent.Agent
+import com.gu.membership.model.{FriendTierPlan, PaidTierPlan, StaffPlan}
+import com.gu.membership.salesforce.Tier.{Partner, Patron, Supporter}
 import com.gu.monitoring.CloudWatchHealth
 import play.api.Logger
+import play.api.libs.concurrent.Akka
+import play.api.libs.concurrent.Execution.Implicits._
 import play.api.mvc.{Action, Controller}
 import services.{GuardianLiveEventService, TouchpointBackend}
+import play.api.Play.current
 
-case class Test(name: String, ok: () => Boolean)
+import scala.concurrent.Future
+
+case class TestCase(name: String, exec: () => Future[Boolean]) {
+  def failedTest() = Logger.warn(s"Test $name failed, health check will fail")
+
+  def ok() = {
+    val result = exec()
+    result.map(passed => if (!passed) failedTest())
+    result.onFailure { case _ => failedTest() }
+    result
+  }
+}
+
+object Test {
+  def apply(name: String, ok: () => Boolean) = TestCase(name, () => Future(ok()))
+}
+
+object ZuoraPing {
+  import com.github.nscala_time.time.Imports._
+  val zuoraSoapService = TouchpointBackend.Normal.zuoraSoapService
+  def ping = zuoraSoapService.lastPingTimeWithin(2.minutes)
+}
 
 object Healthcheck extends Controller {
-  val zuoraService = TouchpointBackend.Normal.zuoraSoapService
+  import scala.concurrent.duration._
+
+  val subscriptionService = TouchpointBackend.Normal.subscriptionService
 
-  val tests = Seq(
+  private val health = Agent(false)
+
+  val productRatePlanTiers = List(FriendTierPlan, StaffPlan,
+    PaidTierPlan(Supporter, true), PaidTierPlan(Supporter, false),
+    PaidTierPlan(Partner, true), PaidTierPlan(Partner, false),
+    PaidTierPlan(Patron, true), PaidTierPlan(Patron, false))
+
+  val tests:Seq[TestCase] = Seq(
     Test("Events", () => GuardianLiveEventService.events.nonEmpty),
     Test("CloudWatch", () => CloudWatchHealth.hasPushedMetricSuccessfully),
-    Test("Zuora", () => zuoraService.lastPingTimeWithin(2.minutes))
-  )
+    Test("ZuoraPing", () => ZuoraPing.ping)) ++
+    productRatePlanTiers.map(tier => TestCase(s"ZuoraCatalog: $tier", () =>
+      subscriptionService.tierPlanRateId(tier).map(_ => true)))
+
+  def updateHealthCheck() =
+    Future.sequence(tests.map(_.ok()))
+      .map { results => health send results.forall(r => r) }
+      .recover { case _ => health send false }
 
-  def healthcheck() = Action { req =>
-    val failedTests = tests.filterNot(_.ok())
+  Akka.system.scheduler.schedule(initialDelay = 0.minutes, interval = 1.minute)(updateHealthCheck())
 
+  def healthcheck() = Action {
     Cached(1) {
-      if (failedTests.nonEmpty) {
-        failedTests.foreach { test =>
-          Logger.warn(s"Test ${test.name} failed, health check will fail")
-        }
-        ServiceUnavailable("Service Unavailable")
-      } else {
-        Ok("OK")
-      }
+      updateHealthCheck()
```

Comment:
Is there a good reason why we need to introduce an agent here? I would keep this code simple and side-effect free by just assuming that these health-check tests get executed whenever matching request comes in